etsi_its_msgs_utils/include/etsi_its_msgs_utils/impl/cpm/cpm_ts_getters.h

```diff
@@ -450,16 +461,35 @@ inline gm::Vector3 getCartesianVelocityOfPerceivedObject(const PerceivedObject &
  * @throws std::invalid_argument If the velocity is no cartesian velocity.
  */
 inline std::tuple<double, double, double> getCartesianVelocityConfidenceOfPerceivedObject(const PerceivedObject &object) {
-  if (!object.velocity_is_present) throw std::invalid_argument("No velocity present in PerceivedObject");
-  if (object.velocity.choice != Velocity3dWithConfidence::CHOICE_CARTESIAN_VELOCITY) {
-    throw std::invalid_argument("Velocity is not Cartesian");
+  if (!object.velocity_is_present) {
+    throw std::invalid_argument("No velocity present in PerceivedObject");
   }
-  double x_confidence = getVelocityComponentConfidence(object.velocity.cartesian_velocity.x_velocity);
-  double y_confidence = getVelocityComponentConfidence(object.velocity.cartesian_velocity.y_velocity);
-  double z_confidence = object.velocity.cartesian_velocity.z_velocity_is_present
-                            ? getVelocityComponentConfidence(object.velocity.cartesian_velocity.z_velocity)
-                            : std::numeric_limits<double>::infinity();
-  return std::make_tuple(x_confidence, y_confidence, z_confidence);
+  if (object.velocity.choice == Velocity3dWithConfidence::CHOICE_POLAR_VELOCITY) {
+    double speed_confidence = getSpeedConfidence(object.velocity.polar_velocity.velocity_magnitude) / etsi_its_msgs::ONE_D_GAUSSIAN_FACTOR;
```

Comment:
Is it correct that the polar values need to be divided by ONE_D_GAUSSIAN_FACTOR, but the cartesian ones not?